exercises/npm-registry/src/package.ts

```diff
@@ -1,22 +1,53 @@
-import { RequestHandler } from 'express';
-import got from 'got';
-import { NPMPackage } from './types';
+import { RequestHandler } from "express";
+import got from "got";
+import { minSatisfying } from "semver";
+import { NPMPackage } from "./types";
 
 /**
  * Attempts to retrieve package data from the npm registry and return it
  */
 export const getPackage: RequestHandler = async function (req, res, next) {
   const { name, version } = req.params;
-
+  const dependencyTree = {};
   try {
     const npmPackage: NPMPackage = await got(
-      `https://registry.npmjs.org/${name}`,
+      `https://registry.npmjs.org/${name}`
     ).json();
 
     const dependencies = npmPackage.versions[version].dependencies;
```

Comment:
This might result in an error if the version provided in `req.params` doesn't exist in the response's version list.
Also, we should validate the request to npm registry returns results and doesn't fail (i.e `req.params.name` is a valid npm package).

I would suggest to first examine the response status for this call and if it's different than 200, we should notify the client by returning 404 - package not found.

```ts
const response = await got(`https://registry.npmjs.org/${name}`);
if (response.status !== 200) {
  res.status(404).send('Package not found in npm registry');
  return;
}
// otherwise - continue
```

Then, we should make sure the version exists:

```suggestion
const versionDeps = npmPackage.versions[version];
if (!versionDeps) {
  res.status(404).send('Package version not found');
  return;
}
const { dependencies } = versionDeps;
```
Now, we know for sure the parameters provided by our client fit our checks. If not, the client knows what he needs to fix in order for the check to complete.